### PR TITLE
Copy new lang assumption to all languages rules

### PR DESCRIPTION
**Pull request by Jym77**
There was a generic feedback in the AG that the assumption in languages rule is unclear. We add some good discussion in #1153 and figured out something which is hopefully better. This copies the new assumption to all the language rules.

I am not sure the new assumption is that good for the "element has valid lang" rule. Input are welcome on this…

Need for Final Call:
This will not require a Final Call (changing an assumption)

---

### **When creating PR:**

- [x] Make sure you're requesting to **pull a branch** (right side) to the `develop` branch (left side).

### **After creating PR:**

- [x] Add yourself (and co-authors) as "Assignees" for PR.
- [x] Add label to indicate if it's a `Rule`, `Definition` or `Chore`.
- [x] Close the issue that the PR resolves (and make sure the issue is referenced in the top of this comment)
- [x] Optionally request feedback from anyone in particular by assigning them as "Reviewers".

## How to Review And Approve

- Go to the “Files changed” tab
- Here you will have the option to leave comments on different lines.
- Once the review is completed, find the “Review changes” button in the top right, select “Approve” (if you are really confident in the rule) or "Request changes" and click “Submit review”.
- Make sure to also review the proposed Final Call period. In case of disagreement, the longer period wins.

### CODE REVIEW

File: _rules/element-lang-valid-de46e4.md [hunk 1]
```diff
@@ -37,7 +37,7 @@ For each test target, the `lang` attribute has a [valid language subtag](#valid-
 
 ## Assumptions
 
-This rule assumes that the presence of a `lang` attribute is being used to comply to WCAG. This rule doesn't test if the attribute is needed to comply to WCAG.
+The language of the page can be set by other methods than the `lang` attribute, for example using HTTP headers or the `meta` element. These methods are not supported by all assistive technologies. This rule assumes that these other methods are insufficient to satisfying [Success Criterion 3.1.2: Language of Parts](https://www.w3.org/TR/WCAG21/#language-of-parts).
```
Comment (dd8) [inline _rules/element-lang-valid-de46e4.md]: Is this relevant to language-of-parts? The other methods are all at the page level (HTTP headers, `meta` element in header, default browser language)

File: _rules/html-page-lang-b5c3f8.md [hunk 2]
```diff
@@ -38,7 +38,7 @@ Each test target has a `lang` attribute that is neither empty ("") nor only [ASC
 
 ## Assumptions
 
-This rule assumes that the presence of a `lang` attribute is being used to comply to WCAG. This rule doesn't test if the attribute is needed to comply to WCAG.
+The language of the page can be set by other methods than the `lang` attribute, for example using HTTP headers or the `meta` element. These methods are not supported by all assistive technologies. This rule assumes that these other methods are insufficient to satisfying [Success Criterion 3.1.1: Language of Page](https://www.w3.org/TR/WCAG21/#language-of-page).
```
Comment (dd8) [inline _rules/html-page-lang-b5c3f8.md]: Looks good

File: _rules/html-page-lang-xml-lang-match-5b7ae0.md [hunk 3]
```diff
@@ -38,7 +38,7 @@ For each test target, the values of the [primary language subtags](https://tools
 
 ## Assumptions
 
-This rule assumes that the presence of a `lang` attribute is being used to comply to WCAG. This rule doesn't test if the attribute is needed to comply to WCAG.
+The language of the page can be set by other methods than the `lang` attribute, for example using HTTP headers or the `meta` element. These methods are not supported by all assistive technologies. This rule assumes that these other methods are insufficient to satisfying [Success Criterion 3.1.1: Language of Page](https://www.w3.org/TR/WCAG21/#language-of-page).
```
Comment (dd8) [inline _rules/html-page-lang-xml-lang-match-5b7ae0.md]: Looks good

File: _rules/element-lang-valid-de46e4.md (hunk 1 above)
Comment (dd8) [reply _rules/element-lang-valid-de46e4.md]: If there's no `lang` attribute on the element, then language of page is relevant since it affects the inherited language

Comment (EmmaJP) [reply _rules/element-lang-valid-de46e4.md]: Perhaps this version simply needs a short sentence at the start to be clear on that relevancy. I don't fully understand the inner browser workings around language determination, but maybe something like:

"If an element lang attribute does not validate, it will default to the language of the page."

Comment (Jym77) [reply _rules/element-lang-valid-de46e4.md]: Yes. I was really not sure about the relevance of this version of the assumption for the "language of parts" rule, but decided to get the discussion going 😄

Comment (carlosapaduarte) [reply _rules/element-lang-valid-de46e4.md]: I agree this only makes sense if prefaced by what @EmmaJP said. Otherwise, it would not be clear how the assumption relates to "Language of Parts".

Comment (dd8) [reply _rules/element-lang-valid-de46e4.md]: If the language is invalid the behaviour is pretty random:

- something it's the inherited lang
- something it's the page lang 
- something it's the system lang

Basically invalid lang means the screen reader picks a language at random (aka undefined behaviour). Could say something like:

"If an element lang attribute does not validate, behaviour is undefined, and content is read in a random language."

Comment (WilcoFiers) [reply _rules/element-lang-valid-de46e4.md]:
```suggestion
The language of the page can be set by other methods than the `lang` attribute; using HTTP headers or the `meta` element. These methods are not known to be supported by assistive technologies. This rule assumes that these other methods are insufficient to satisfying [Success Criterion 3.1.2: Language of Parts](https://www.w3.org/TR/WCAG21/#language-of-parts).
```

Comment (Jym77) [reply _rules/element-lang-valid-de46e4.md]: There was already a new assumption for "language of element" worked in #1133 that was merged today. So I just kept it 😇

File: _rules/html-page-lang-b5c3f8.md (hunk 2 above)
Comment (WilcoFiers) [inline _rules/html-page-lang-b5c3f8.md]: Still, these aren't known to be supported by **any** AT. Saying "are not supported by all assistive technologies" implies that it can be used. If we can't name a single one that supports it, we should be clear about that. Otherwise this should be an accessibility support note.

Comment (Jym77) [reply _rules/html-page-lang-b5c3f8.md]: According to tests done by @dd8 in https://github.com/act-rules/act-rules.github.io/issues/668#issuecomment-534606384 there methods are exposed to the accessibility tree by **some** browsers, so it will be available to **some** combinations of UA+AT.
